CONTRIBUTING.md

```diff
@@ -3,9 +3,200 @@
+  - [Exact Steps](#exact-steps)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
 ## Certificate of Origin
 
 By contributing to this project you agree to the Developer Certificate of
 Origin (DCO). This document was created by the Linux Kernel community and is a
 simple statement that you, as a contributor, have the legal right to make the
 contribution. See the [DCO](DCO) file for details.
+
+## Finding Things That Need Help
+
+If you're new to the project and want to help, but don't know where to start, we
+have a semi-curated list of issues that
+should not need deep knowledge of the system. [Have a look and see if anything
+sounds interesting](https://github.com/metal3-io/baremetal-operator/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22).
+Alternatively, read some of the docs on other controllers and try to write your
+own, file and fix any/all issues that come up, including gaps in documentation!
+
+## Versioning
+
+### Codebase
+
+Baremetal Operator doesn't follow release cadence and versioning of upstream Kubernetes semantic versioning. This codebase guarantees the following:
+
+- A (*minor*) release CAN include:
+  - Introduction of new API versions, or new Kinds.
+  - Compatible API changes like field additions, deprecation notices, etc.
+  - Breaking API changes for deprecated APIs, fields, or code.
+  - Features, promotion or removal of feature gates.
+  - And more!
+
+- A (*patch*) release SHOULD only include backwards compatible set of bugfixes.
+
+These guarantees extend to all code exposed in our Go Module, including
+*types from dependencies in public APIs*.
+Types and functions not in public APIs are not considered part of the guarantee.
+
+#### Backporting
+
+We only accept backports of critical bugs, security issues, or bugs without easy workarounds, any
+backport MUST not be breaking for either API or behavioral changes.
+We generally do not accept PRs against older release branches.
+
+## Branches
+
+Baremetal Operator has only the *main* branch currently.
+
+The goal is to have two types of branches in future: the *main* branch and *release-X* branches.
+
+The *main* branch is where development happens. All the latest and
+greatest code, including breaking changes, happens on main. Currently patch releases are done from this branch
+only.
+
+The *release-X* branches will contain stable, backwards compatible code. On every
+major or minor release, a new branch will be created. It is from these
+branches that minor and patch releases will be tagged. In some cases, it may
+be necessary to open PRs for bugfixes directly against stable branches, but
+this should generally not be the case.
+
+## Contributing a Patch
+
+1. If you haven't already done so, sign a Contributor License Agreement (see
+   details above).
+1. Fork the desired repo, develop and test your code changes.
+1. Submit a pull request.
+
+All code PR must be labeled with one of
```

Comment:
I don't think most of us follow this and, while I find it useful, I'd not use "must" here.